analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -63,16 +70,131 @@ public sealed class MemberInitializerRedundant : SonarDiagnosticAnalyzer
                     {
-                        InitializationVerifierFactory.InstanceMemberInitializationVerifier.Report(c, declaration, members);
+                        InitializationVerifierFactory.InstanceMemberInitializationVerifier.Report(c, declaration, members, forceSonarCfg);
                     }
-                    InitializationVerifierFactory.StaticMemberInitializationVerifier.Report(c, declaration, members);
+                    InitializationVerifierFactory.StaticMemberInitializationVerifier.Report(c, declaration, members, forceSonarCfg);
                 },
                 // For record support, see details in https://github.com/SonarSource/sonar-dotnet/pull/4756
                 // it is difficult to work with instance record constructors w/o raising FPs
                 SyntaxKind.ClassDeclaration,
                 SyntaxKind.StructDeclaration,
                 SyntaxKind.InterfaceDeclaration);
 
+        private class RedundancyChecker
+        {
+            private readonly ISymbol memberToCheck;
+            private readonly SemanticModel semanticModel;
+
+            public RedundancyChecker(ISymbol memberToCheck, SemanticModel semanticModel)
+            {
+                this.memberToCheck = memberToCheck;
+                this.semanticModel = semanticModel;
+            }
+
+            public bool IsMemberUsedInsideLambda(SyntaxNode instruction) =>
+                instruction.DescendantNodes()
+                    .OfType<IdentifierNameSyntax>()
+                    .Select(GetPossibleMemberAccessParent)
+                    .Any(IsMatchingMember);
+
+            public static ExpressionSyntax GetPossibleMemberAccessParent(SyntaxNode node)
+            {
+                if (node is MemberAccessExpressionSyntax memberAccess)
+                {
+                    return memberAccess;
+                }
+                else
+                {
+                    return GetPossibleMemberAccessParent(node as IdentifierNameSyntax);
+                }
+            }
+
+            public bool IsMatchingMember(ExpressionSyntax expression)
+            {
+                IdentifierNameSyntax identifier = null;
+
+                if (expression.IsKind(SyntaxKind.IdentifierName))
+                {
+                    identifier = (IdentifierNameSyntax)expression;
+                }
+
+                if (expression is MemberAccessExpressionSyntax memberAccess
+                    && memberAccess.Expression.IsKind(SyntaxKind.ThisExpression))
+                {
+                    identifier = memberAccess.Name as IdentifierNameSyntax;
+                }
+
+                if (expression is ConditionalAccessExpressionSyntax conditionalAccess
+                    && conditionalAccess.Expression.IsKind(SyntaxKind.ThisExpression))
+                {
+                    identifier = (conditionalAccess.WhenNotNull as MemberBindingExpressionSyntax)?.Name as IdentifierNameSyntax;
+                }
```

Comment:
Let's move this identifier-searching part to a local method `ExtractIdentifier` with `if return else if return else if return else return null` structure.
After that, this whole thing could be somethign like
```
return ExtractIdentifier() is identifer
    && semanticModel.GetSymbolInfo(identifier).Symbol is { } symbol
    && memberToCheck.Equals(assignedSymbol);
```
This improvement probably only makes sense if we'll use this in the Operation-base version.